tests/src/Context/WhatsNewContext.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\joinup\Context;
+
+use Drupal\DrupalExtension\Context\RawDrupalContext;
+use PHPUnit\Framework\Assert;
+
+/**
+ * Behat step definitions and related methods provided by the whats_new module.
+ */
+class WhatsNewContext extends RawDrupalContext {
+
+  /**
+   * Asserts that the support menu button has the "whats_new" class attached.
+   *
+   * @Then I should see the bell icon in the support menu
+   */
+  public function assertSupportMenuWhatsNewClass(): void {
+    $element = $this->getSession()->getPage()->findById('support-menu__button');
+    Assert::assertTrue($element->hasClass('whats_new'));
+  }
+
+  /**
+   * Asserts that the support menu does not have the "whats_new" class attached.
+   *
+   * @Then I should not see the bell icon in the support menu
+   */
+  public function assertSupportMenuNotWhatsNewClass(): void {
+    $element = $this->getSession()->getPage()->findById('support-menu__button');
+    Assert::assertFalse($element->hasClass('whats_new'));
+  }
+
+  /**
+   * Asserts that the support menu button has the "whats_new" class attached.
+   *
+   * @param string $link_title
+   *   The link title.
+   *
+   * @Then the :link_title link should be featured as what's new
+   */
+  public function assertSupportMenuLinkWhatsNewClass(string $link_title): void {
+    foreach ($this->getFeaturedSupportLinks() as $element) {
+      if ($element->getText() === $link_title) {
+        return;
+      }
+    }
+    throw new \Exception("No link with title '{$link_title}' has been found marked as featured in the support menu.");
```

Comment:
This makes the IDE yell that docblock contains no`@throws` tag. And IDE is right because we should start doing better with our custom step definition. When testing, there's a big difference between a test ending with **error** vs. **failed**. From https://phpunit.readthedocs.io/en/9.2/textui.html:

> PHPUnit distinguishes between failures and errors. A failure is a violated PHPUnit assertion such as a failing `assertSame()` call. An error is an unexpected exception or a PHP error. Sometimes this distinction proves useful since errors tend to be easier to fix than failures.

So, when the expectation is not met we should use `ExpectationFailedException`. Or when an element that we want to test is missing, we should use `ElementNotFoundException`. This allows better understanding of what's going on as...

> If you have a big list of problems, it is best to tackle the errors first and see if you have any failures left when they are all fixed.